main.go

```diff
@@ -88,6 +91,20 @@ func main() {
 		sugarLogger.Fatal("must specify the GIT_PROVIDER env variable (i.e. cache, memory)")
 	}
 
-	pizzaOvenServer := server.NewPizzaOvenServer(pizzaOven, pizzaGitProvider, sugarLogger)
+	// Initializes configuration using a provided yaml file
+	var config server.Config
+	configFile, err := os.ReadFile(configPath)
+	if err != nil {
+		sugarLogger.Debugf("No yaml configuration file was provided")
+	} else {
+		err = yaml.Unmarshal(configFile, &config)
+		if err != nil {
+			sugarLogger.Debugf("Could not unmarshal the config file")
```

Comment:
If the user has designated the config flag, this should fail and not just print a warning. That will be much faster feedback for misconfigurations.

Reply:
Recent change errors immediately and returns if a config was provided and could not be parsed.

```go
                err = yaml.Unmarshal(configFile, &configParser)
		if err != nil {
			sugarLogger.Errorf("Could not unmarshal the config file: %v", err)
			return
		}
```